libs/remix-ui/workspace/src/lib/actions/workspace.ts

```diff
@@ -75,29 +75,40 @@ export const loadWorkspacePreset = async (template: 'gist-template' | 'code-temp
 
   switch (template) {
     case 'code-template':
-      // creates a new workspace code-sample and loads code from url params.
-      try {
-        let path = ''; let content = ''
-
-        if (params.code) {
-          const hash = bufferToHex(keccakFromString(params.code))
+    // creates a new workspace code-sample and loads code from url params.
+    try {
+      let path = ''; let content = ''
 
-          path = 'contract-' + hash.replace('0x', '').substring(0, 10) + '.sol'
-          content = atob(params.code)
-          workspaceProvider.set(path, content)
-        }
-        if (params.url) {
-          const data = await plugin.call('contentImport', 'resolve', params.url)
+      if (params.code) {
+        const hash = bufferToHex(keccakFromString(params.code))
 
-          path = data.cleanUrl
-          content = data.content
+        path = 'contract-' + hash.replace('0x', '').substring(0, 10) + '.sol'
+        content = atob(params.code)
+        workspaceProvider.set(path, content)
+      }
+      if (params.url) {
+        const data = await plugin.call('contentImport', 'resolve', params.url)
+
+        path = data.cleanUrl
+        content = data.content
+        if (typeof content === 'object') {
+          const standardInput = content as JSONStandardInput
+          if (standardInput.language && standardInput.language === "Solidity" && standardInput.sources) {
+            const obj = standardInput.sources as any
```

Comment:
`obj` is defined here, but i can't find where it is being used. You can remove this variable declaration if it is not needed.

Reply:
Thanks, fixed!